amethyst_rendy/src/camera.rs

```diff
@@ -388,44 +482,99 @@ impl Projection {
         }
     }
 
+    /// Returns a reference to this `Projection` as [CustomMatrix] if it is in fact an `CustomMatrix`
+    /// projection. Otherwise, `None` is returned.
+    pub fn as_custom_matrix(&self) -> Option<&CustomMatrix> {
+        match *self {
+            Projection::CustomMatrix(ref s) => Some(s),
+            _ => None,
+        }
+    }
+
+    /// Returns a reference to this `Projection` as [CustomMatrix] if it is in fact an `CustomMatrix`
+    /// projection. Otherwise, `None` is returned.
+    pub fn as_custom_matrix_mut(&mut self) -> Option<&mut CustomMatrix> {
+        match *self {
+            Projection::CustomMatrix(ref mut s) => Some(s),
+            _ => None,
+        }
+    }
+
     /// Returns a reference to the inner matrix represpentation of this projection.
     pub fn as_matrix(&self) -> &Matrix4<f32> {
         match *self {
             Projection::Orthographic(ref s) => s.as_matrix(),
             Projection::Perspective(ref s) => s.as_matrix(),
+            Projection::CustomMatrix(ref s) => s.as_matrix(),
+        }
+    }
+
+    /// Returns a reference to the inner inverse matrix represpentation of this projection.
+    pub fn as_inverse_matrix(&self) -> &Matrix4<f32> {
+        match *self {
+            Projection::Orthographic(ref s) => s.as_inverse_matrix(),
+            Projection::Perspective(ref s) => s.as_inverse_matrix(),
+            Projection::CustomMatrix(ref s) => s.as_inverse_matrix(),
+        }
+    }
+
+    /// Returns the near-clip value of the current camera.
+    pub fn near(&self) -> f32 {
+        match *self {
+            Projection::Orthographic(ref s) => s.near(),
+            Projection::Perspective(ref s) => s.near(),
+            Projection::CustomMatrix(ref s) => {
+                panic!("Custom Matrix does not support near and far field projection!")
```

Comment:
This is still not great. The existance of such API might mean that you can't use the `CustomMatrix` variant in practice, because some code unknowingly uses those methods. The `fn near` and `fn far` should either be removed or some general equasion should be applied on the `CustomMatrix` variant. If it's removed, you can always match on your specific camera type you are interested in and get the value out.

Note that `fn far` might soon not even work on `Perspective` (or at least not on every one), as there is such thing as "infinite far plane".